spotbugs/src/main/java/edu/umd/cs/findbugs/detect/ConstructorThrow.java

```diff
@@ -0,0 +1,152 @@
+package edu.umd.cs.findbugs.detect;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.apache.bcel.Const;
+
+import edu.umd.cs.findbugs.BugInstance;
+import edu.umd.cs.findbugs.BugAccumulator;
+import edu.umd.cs.findbugs.BugReporter;
+import edu.umd.cs.findbugs.bcel.OpcodeStackDetector;
+import org.apache.bcel.classfile.Method;
+import org.apache.bcel.classfile.ExceptionTable;
+import org.apache.bcel.classfile.JavaClass;
+
+/**
+ * This detector can find constructors that throw exception.
+ */
+public class ConstructorThrow extends OpcodeStackDetector {
+
+    private final BugReporter bugReporter;
+    private final BugAccumulator bugAccumulator;
+    private final Set<String> calledFromCtor = new HashSet<String>();
+
+    private boolean isFinalClass = false;
+    private boolean isFinalFinalizer = false;
+    private boolean isFirstPass = true;
+
+    public ConstructorThrow(BugReporter bugReporter) {
+        this.bugReporter = bugReporter;
+        this.bugAccumulator = new BugAccumulator(bugReporter);
+    }
+
+    /**
+     * Visit a class to find the constructor, then collect all the methods that gets called in it.
+     * Also we are checking for final declaration on the class, or a final finalizer, as if present
+     * no finalizer attack can happen.
+     */
+    @Override
+    public void visit(JavaClass obj) {
+        resetState();
+        if (obj.isFinal()) {
+            isFinalClass = true;
+            return;
+        }
+        for (Method m : obj.getMethods()) {
+            // First visit the constructor, it might not be at the start of the class.
+            if ("<init>".equals(m.getName())) {
```

Comment:
This method name should probably be a constant somewhere (`Const.CONSTRUCTOR_NAME` perhaps?).

Reply:
I think this conversation can be resolved, since it got fixed in https://github.com/spotbugs/spotbugs/pull/2095/commits/10a906babe2a55c312203901536b6b43f09764ba.